moved colony switcher to separate componentn and add category to colo…

KatarzynaStefanowiczPagepro opened the pull request:
…ny dropdown

## Description

- [] TODO write description

**New stuff** ✨

* New stuff goes here

**Changes** 🏗

* Changes go here

**Deletions** ⚰️

* Deletions go here

## TODO

- [ ] Add TODOs

(Resolves | Contributes to) #31415

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/JoinColony/colonyCDapp?pullRequest=859) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/JoinColony/colonyCDapp?pullRequest=859) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/JoinColony/colonyCDapp?pullRequest=859) it.</sub>

KatarzynaStefanowiczPagepro commented:
https://tw.auroracreation.com/app/tasks/15516242

![image](https://github.com/JoinColony/colonyCDapp/assets/119701775/94d01028-445c-46ff-bc7a-62f0ac6b8502)

- Moved colony switcher to a separate component
- Added a category to colony dropdown

CzarekDryl commented on src/components/common/Extensions/ColonySwitcher/partials/ColoniesDropdown.tsx:
```diff
@@ -17,17 +18,42 @@ const ColoniesDropdown: FC<ColoniesDropdownProps> = ({
-      return group;
-    },
-    {},
+  const updatedWatchList = useMemo(
+    () =>
+      watchlist.map((item) => {
+        const newNetwork = Object.keys(NETWORK_DATA).find(
+          (network) =>
+            NETWORK_DATA[network].chainId ===
+            item?.colony.chainMetadata.chainId,
+        );
+
+        if (!newNetwork) {
+          return item;
+        }
+
+        return {
+          ...item,
+          colony: {
+            chainMetadata: {
+              chainId: item?.colony.chainMetadata.chainId,
+              network: newNetwork,
+            },
+            colonyAddress: item?.colony.colonyAddress,
+            name: item?.colony.name,
+          },
+        };
+      }),
+    [watchlist],
   );
 
+  const groupByCategory = updatedWatchList.reduce((group, item) => {
+    const network = (item && item.colony.chainMetadata?.network) || '';
+    // eslint-disable-next-line no-param-reassign
+    group[network] = group[network] ?? [];
+    group[network].push(item);
+    return group;
+  }, {});
```
Can you move this to hooks?

KatarzynaStefanowiczPagepro replied:
Done
